### ISSUE TITLE
Virtual Machine could not be started

### ISSUE THREAD

**Issue by daPhipz** [kubuntu-team/KubuQA#2]
Hello!
I tried to spin up a test environment using this tool, but it failed.

### Steps to reproduce:
1. Download and run the script as per README instructions (script location is `~/Downloads`)
2. Run the script. It prompted me to install some missing packages, as stated in the README. I authorized the installation each time and everything went according to plan.
3.  When prompted to select the boot medium, I chose `ISO`. Then, the following error message appeared:
![Screenshot_20240413_121950](https://github.com/kubuntu-team/KubuQA/assets/30795174/0de84337-636e-4036-ad63-f78423895cd4)
Translating loosely to:
> The virtual machine could not be started. This could be caused by a missing Operating System or a wong boot order. Mountng a OS installation DVD could solve this problem. If a ISO file will be selected, we will try to mount it after this dialog has been closed.
4. I then enter the path to the ISO file the script downloaded earlier (`~/Downloads/KubuntuTestISO/noble-desktop-amd64.iso`), I eventually land at the same error message.

Here is all command line output the script produced:

```
VBoxManage could not be found, attempting to install.
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Die folgenden zusätzlichen Pakete werden installiert:
  libgsoap-2.8.124 liblzf1 libtpms0 libvncserver1 virtualbox-dkms virtualbox-qt
Vorgeschlagene Pakete:
  vde2 virtualbox-guest-additions-iso
Die folgenden NEUEN Pakete werden installiert:
  libgsoap-2.8.124 liblzf1 libtpms0 libvncserver1 virtualbox virtualbox-dkms virtualbox-qt
0 aktualisiert, 7 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
Es müssen noch 0 B von 54,5 MB an Archiven heruntergeladen werden.
Nach dieser Operation werden 194 MB Plattenplatz zusätzlich benutzt.
Vormals nicht ausgewähltes Paket libgsoap-2.8.124:amd64 wird gewählt.
(Lese Datenbank ... 245359 Dateien und Verzeichnisse sind derzeit installiert.)
Vorbereitung zum Entpacken von .../0-libgsoap-2.8.124_2.8.124-2_amd64.deb ...
Entpacken von libgsoap-2.8.124:amd64 (2.8.124-2) ...
Vormals nicht ausgewähltes Paket liblzf1:amd64 wird gewählt.
Vorbereitung zum Entpacken von .../1-liblzf1_3.6-4_amd64.deb ...
Entpacken von liblzf1:amd64 (3.6-4) ...
Vormals nicht ausgewähltes Paket libtpms0:amd64 wird gewählt.
Vorbereitung zum Entpacken von .../2-libtpms0_0.9.3-0ubuntu2_amd64.deb ...
Entpacken von libtpms0:amd64 (0.9.3-0ubuntu2) ...
Vormals nicht ausgewähltes Paket libvncserver1:amd64 wird gewählt.
Vorbereitung zum Entpacken von .../3-libvncserver1_0.9.14+dfsg-1_amd64.deb ...
Entpacken von libvncserver1:amd64 (0.9.14+dfsg-1) ...
Vormals nicht ausgewähltes Paket virtualbox-dkms wird gewählt.
Vorbereitung zum Entpacken von .../4-virtualbox-dkms_7.0.14-dfsg-1~ubuntu1.23.10.1_amd64.deb ...
Entpacken von virtualbox-dkms (7.0.14-dfsg-1~ubuntu1.23.10.1) ...
Vormals nicht ausgewähltes Paket virtualbox wird gewählt.
Vorbereitung zum Entpacken von .../5-virtualbox_7.0.14-dfsg-1~ubuntu1.23.10.1_amd64.deb ...
Entpacken von virtualbox (7.0.14-dfsg-1~ubuntu1.23.10.1) ...
Vormals nicht ausgewähltes Paket virtualbox-qt wird gewählt.
Vorbereitung zum Entpacken von .../6-virtualbox-qt_7.0.14-dfsg-1~ubuntu1.23.10.1_amd64.deb ...
Entpacken von virtualbox-qt (7.0.14-dfsg-1~ubuntu1.23.10.1) ...
libgsoap-2.8.124:amd64 (2.8.124-2) wird eingerichtet ...
libvncserver1:amd64 (0.9.14+dfsg-1) wird eingerichtet ...
liblzf1:amd64 (3.6-4) wird eingerichtet ...
libtpms0:amd64 (0.9.3-0ubuntu2) wird eingerichtet ...
virtualbox-dkms (7.0.14-dfsg-1~ubuntu1.23.10.1) wird eingerichtet ...
Loading new virtualbox-7.0.14 DKMS files...
Building for 6.5.0-27-generic
Building initial module for 6.5.0-27-generic
Done.

vboxdrv.ko.zst:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/6.5.0-27-generic/updates/dkms/

vboxnetadp.ko.zst:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/6.5.0-27-generic/updates/dkms/

vboxnetflt.ko.zst:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/6.5.0-27-generic/updates/dkms/
depmod....
virtualbox (7.0.14-dfsg-1~ubuntu1.23.10.1) wird eingerichtet ...
virtualbox-qt (7.0.14-dfsg-1~ubuntu1.23.10.1) wird eingerichtet ...
Trigger für mailcap (3.70+nmu1ubuntu1) werden verarbeitet ...
Trigger für desktop-file-utils (0.26-1ubuntu5) werden verarbeitet ...
Trigger für hicolor-icon-theme (0.17-2) werden verarbeitet ...
Trigger für libc-bin (2.38-1ubuntu6.1) werden verarbeitet ...
Trigger für man-db (2.11.2-3) werden verarbeitet ...
Trigger für shared-mime-info (2.2-1) werden verarbeitet ...
./KubuQA.sh: line 82: check_existing_vm: command not found
--2024-04-13 10:43:03--  https://cdimages.ubuntu.com/kubuntu/daily-live/current/noble-desktop-amd64.iso
Resolving cdimages.ubuntu.com (cdimages.ubuntu.com)... 185.125.190.37, 91.189.91.124, 185.125.190.40, ...
Connecting to cdimages.ubuntu.com (cdimages.ubuntu.com)|185.125.190.37|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4172765184 (3,9G) [application/x-iso9660-image]
Saving to: ‘noble-desktop-amd64.iso’

noble-desktop-amd64.iso      100%[==============================================>]   3,89G  2,02MB/s    in 31m 5s  

2024-04-13 11:14:08 (2,13 MB/s) - ‘noble-desktop-amd64.iso’ saved [4172765184/4172765184]

Virtual machine 'TestKubuntuInstall' is created and registered.
UUID: b7a8ac9b-d86b-4a76-b6fb-5f0549c2bb71
Settings file: '/home/philipp/VirtualBox VMs/TestKubuntuInstall/TestKubuntuInstall.vbox'
VBoxManage: error: Could not find file for the medium '/home/philipp/VirtualBox VMs/TestKubuntuInstall/TestKubuntuInstall.vdi' (VERR_FILE_NOT_FOUND)
VBoxManage: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MediumWrap, interface IMedium, callee nsISupports
VBoxManage: error: Context: "OpenMedium(Bstr(pszFilenameOrUuid).raw(), enmDevType, enmAccessMode, fForceNewUuidOnOpen, pMedium.asOutParam())" at line 197 of file VBoxManageDisk.cpp
VBoxManage: error: Invalid UUID or filename "/home/philipp/VirtualBox VMs/TestKubuntuInstall/TestKubuntuInstall.vdi"
Waiting for VM "TestKubuntuInstall" to power on...
VM "TestKubuntuInstall" has been successfully started.
```

> `./KubuQA.sh: line 82: check_existing_vm: command not found`

This line seems to be an error, maybe it has something to do with it?

**Comment by daPhipz** [comment]
>  ./KubuQA.sh: line 82: check_existing_vm: command not found
This line seems to be an error, maybe it has something to do with it?

It seems that this is unrelated, since my problem still persists even with the changes done in #3.

**Comment by ricktimmis** [comment]
Hi @daPhipz thank you so much for trying out the tool. I'm afraid this is my fault! I changed the function definition to  check_exiisting{} and didn't update the call on line 82.
If you change check_existing_vm on line 82 to check_existing that should fix it.
I'm on my tablet right now, so not so easy to do, if you want to push another PR I would be very grateful.
Best wishes 
Rick

**Comment by daPhipz** [comment]
No problem, I stumbled upon it reading the Beta announcement for Kubuntu 24.04!
I fixed the method call in #3 and even with these changes applied, it fails.

Something I found: When setting up a `noble` VM manually in VirtualBox, two Mass Storage Devices get created: one optical drive, holding the ISO that gets booted, and one SATA device that holds the vdi.

When using the script and selecting to boot from ISO, only the vdi file shows up.
I'm sorry that I don't know much about the `VBoxManage` tool, it's the first time I see that you can interact with VirtualBox via the command line! :sweat_smile:

**Comment by daPhipz** [comment]
This is what I mean:
![Screenshot_20240413_193724](https://github.com/kubuntu-team/KubuQA/assets/30795174/6d0541b0-ee17-41a5-8a69-47d1b9e207cb)
![Screenshot_20240413_193737](https://github.com/kubuntu-team/KubuQA/assets/30795174/a39d170d-4bc3-4ca1-976f-11666cbfb3f4)
